store/iavl/store.go

```diff
@@ -275,31 +277,42 @@ func (st *Store) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 			break
 		}
 
-		if req.Prove {
-			value, proof, err := tree.GetVersionedWithProof(key, res.Height)
-			if err != nil {
-				res.Log = err.Error()
-				break
-			}
-			if proof == nil {
-				// Proof == nil implies that the store is empty.
-				if value != nil {
-					panic("unexpected value for an empty proof")
-				}
+		_, res.Value = tree.GetVersioned(key, res.Height)
```

Comment:
Note we get the value from res.Height, but below use the tree from the current height.
This will fail if the key changed between the querier and current heights.